pettingzoo/butterfly/pistonball/pistonball.py

```diff
@@ -523,7 +526,7 @@ def render(self, mode="human"):
         )
 
     def step(self, action):
-        if self.dones[self.agent_selection]:
+        if self.terminations[self.agent_selection]:
```

Comment:
This needs to be `if self.terminations[self.agent_selection] or self.truncations[self.agent_selection]` I think.